src/main/java/com/iexec/worker/compute/pre/PreComputeService.java

```diff
@@ -81,6 +84,30 @@ public boolean runStandardPreCompute(TaskDescription taskDescription) {
 
     public String runTeePreCompute(TaskDescription taskDescription, WorkerpoolAuthorization workerpoolAuth) {
         String chainTaskId = taskDescription.getChainTaskId();
+        // verify enclave configuration for compute stage
+        TeeEnclaveConfiguration enclaveConfig = taskDescription.getAppEnclaveConfiguration();
+        if (StringUtils.isEmpty(enclaveConfig.getFingerprint())
+                || BytesUtils.stringToBytes(enclaveConfig.getFingerprint()).length != 32) {
```

Comment:
Maybe in a future PR we can add something like `verifyFingerprint()` to `TaskDescription` directly.

Reply:
Thank you!